api/CcsSso.Core.Service/External/OrganisationGroupService.cs

```diff
@@ -886,5 +928,115 @@ private async Task GeneratePendingRequests(OrganisationUserGroup group, List<Use
       }
     }
 
+    private async Task ModifyUserRoles(OrganisationGroupRequestInfo organisationGroupRequestInfo, int organisationId)
+    {
+      var orgAdminAccessRoleId =await _externalOrgHelperService.GetOrganisationAdminAccessRoleId(organisationId);
+      if (organisationGroupRequestInfo.UserInfo?.AddedUserIds.Count > 0)
+      {
+        await AddAdminRole(organisationGroupRequestInfo, orgAdminAccessRoleId, organisationId);
+      }
+      if (organisationGroupRequestInfo.UserInfo?.RemovedUserIds.Count > 0)
+      {
+        await RemoveAdminRole(organisationGroupRequestInfo, organisationId);
+      }
+    }
+
+    private async Task RemoveAdminRole(OrganisationGroupRequestInfo organisationGroupRequestInfo, int organisationId)
+    {
+      var groupUpdatingUsers = await GetUsers(organisationGroupRequestInfo.UserInfo.RemovedUserIds, organisationId);
+
+      if (groupUpdatingUsers == null)
+      {
+        throw new ResourceNotFoundException();
+      }
+      //Cannot remove myprofile from the admin group 
+      if (groupUpdatingUsers.Any(x => x.Id == _requestContext.UserId))
+      {
+        throw new CcsSsoException(ErrorConstant.ErrorCannotRemoveMyProfile);
+      }
+      // Remove the admin access role from the user
+      foreach (var user in groupUpdatingUsers)
+      {
+        if (user.UserAccessRoles != null)
+        {
+          var adminAccessRoleInfo = user.UserAccessRoles.FirstOrDefault(uar => !uar.IsDeleted 
+           && uar.OrganisationEligibleRole.CcsAccessRole.CcsAccessRoleNameKey == Contstant.OrgAdminRoleNameKey);
+          if (adminAccessRoleInfo != null)
+          {
+            adminAccessRoleInfo.IsDeleted = true;
+          }
+        }
+      }
+    }
+
+    private async Task AddAdminRole(OrganisationGroupRequestInfo organisationGroupRequestInfo, int orgRoleId, int organisationId)
+    {
+      var groupUpdatingUsers = await GetUsers(organisationGroupRequestInfo.UserInfo.AddedUserIds, organisationId);
+      if (groupUpdatingUsers == null)
+      {
+        throw new ResourceNotFoundException();
+      }
+      if (groupUpdatingUsers.Any(x => x.Id == _requestContext.UserId))
+      {
+        throw new CcsSsoException(ErrorConstant.ErrorCannotAddMyProfile);
+      }
+      foreach (var user in groupUpdatingUsers)
+      {
+        if (user.UserAccessRoles == null)
+        {
+          user.UserAccessRoles = new List<UserAccessRole>();
+        }
+        if (user.UserAccessRoles.Any(x => x.OrganisationEligibleRoleId == orgRoleId && !x.IsDeleted))
+        {
+          throw new CcsSsoException(ErrorConstant.ErrorInvalidRoleInfo);
+        }
+        user.UserAccessRoles.Add(new UserAccessRole
+        {
+          UserId = user.Id,
+          OrganisationEligibleRoleId = orgRoleId
+        }); ;
+      }
+    }
+
+    private async Task CheckInvalidRoleInfo(List<int> roleIds)
+    {
+      var serviceRoleGroups = await _serviceRoleGroupMapperService.OrgRolesToServiceRoleGroupsAsync(roleIds);
```

Comment:
we don't want to user `_serviceRoleGroupMapperService.OrgRolesToServiceRoleGroupsAsync(`